lib/vip-encloseme-cleanup.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+class VIP_Encloseme_Cleanup {
+    const OPTION_NAME = 'vip_encloseme_cleanup';
+    const CRON_HOOK = 'vip_encloseme_cleanup_hook';
+    const CRON_INTERVAL = 'vip_cleanup_interval';
+
+    public static function init() {
+        add_filter( 'cron_schedules', [ __CLASS__, 'add_cron_schedule' ] );
+        add_action( 'init', [ __CLASS__, 'schedule_cleanup' ], 99999 );
+        add_action( self::CRON_HOOK, [ __CLASS__, 'cleanup_encloseme_meta' ] );
+    }
+
+    public static function add_cron_schedule( $schedule ) {
+        if ( isset( $schedule[ self::CRON_INTERVAL ] ) ) {
+            return $schedule;
+        }
+
+        $schedule[ self::CRON_INTERVAL ] = [
+			'interval' => 1200,
+			'display' => __( 'Once every twenty minutes.' ),
+        ];
+        
+        return $schedule;
+    }
+
+    public static function schedule_cleanup() {
+        $already_ran = get_option( self::OPTION_NAME );
+
+        if ( $already_ran && wp_next_scheduled( self::CRON_HOOK ) ) {
+            wp_clear_scheduled_hook( self::CRON_HOOK );
+        }
+
+        if ( false === $already_ran && ! wp_next_scheduled( self::CRON_HOOK ) ) {
+            wp_schedule_event( time(), self::CRON_INTERVAL, self::CRON_HOOK );
+        }
+    }
+
+    public static function cleanup_encloseme_meta() {
+        if ( ! wp_doing_cron() ) {
+            return;
+        }
+
+        global $wpdb;
+        $find_meta_query = "SELECT post_id FROM $wpdb->postmeta WHERE meta_key = '_encloseme' LIMIT 1"; // We use this instead of count to mitigate the risk of the SQL server going away. 
+        $meta_exists = $wpdb->get_var( $find_meta_query ); 
+
+        if ( is_null( $meta_exists ) ) {
+            update_option( self::OPTION_NAME, time() );
+            wpcom_vip_irc( '#vip-go-encloseme-meta-cleanup', sprintf( 'No _encloseme meta cleanup needed for %s!', get_site_url() ) );
+            return;
+        }
+
+        wpcom_vip_irc( '#vip-go-encloseme-meta-cleanup', sprintf( 'Starting _encloseme meta cleanup for %s.', get_site_url() ) );
+
+        do {
+            $pids = $wpdb->get_results(
+                "SELECT DISTINCT post_id FROM $wpdb->postmeta WHERE meta_key = '_encloseme' LIMIT 1000",
+                ARRAY_N
+            );
+            $pids = array_map( function( $pid ) {
+                return $pid[0];
+            }, $pids );
+            
+            foreach( $pids as $pid ) {
+                delete_post_meta( $pid, '_encloseme' );
+                if ( defined( 'DISABLE_VIP_CLEANUP_ENV' ) && true === DISABLE_VIP_CLEANUP_ENV ) {
+                    break 2;
+                }
+            }
+    
+            sleep( 3 );
+
+            $meta_exists = $wpdb->get_var( $find_meta_query );
+        } while ( ! is_null( $meta_exists ) && false === defined( 'DISABLE_VIP_CLEANUP_ENV' ) );
```

Comment:
Infinite loops are scary :) I'm also not sure we need it with the cron-based approach?

If we do want to process a larger batch size per cron event, it may be better to use a `for` loop:

```
// 5 is arbitrary here
for ( $count = 0; $count < 5; $count++ ) {
    // check constant and bail if not set

    // get 1000 entries

   // if not found, bail

   // cleanup

   // sleep
}
```

Reply:
@mjangda What about for sites that range from either 100 rows that need to be deleted versus 500,000+ rows that need to be deleted? I'm not sure what we would set the batch size to with such different scenarios.  Or would we just set it to a very high number?

Reply:
We can set it high enough and frequent enough so that it doesn't take days/months to deal with really large sites but not too high or too frequent to cause issues.

For example, If we process 10K entries per event, we'll be finished a 500K site in ~16 hours (`500000 / 10000 / 3` -- 3 because we're running 3 times an hour). We can drop the frequency to 15 mins and be done in 12.5 hours instead.